echo_server.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+
+
+def server_socket_function():
+    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+            message = conn.recv(32)
+            if message:
+                conn.sendall("I recieved your message. Stop talking to me. You are annoying.")
+    except KeyboardInterrupt:
+        conn.close()
```

Comment:
You've got a couple of problems here:

1.  You never receive more than 32 bytes of the message sent by the client.  What if the message is longer than that?
2.  You don't echo the message sent by the client back to the client unaltered, which means you can't tell that your server isn't working.
3.  You never close the server socket so it can be recycled by the OS.  

Now that I see this, the problems you are having with it having worked once and then not again make sense to me.  You're leaving behind a pile of unclosed, unreachable sockets.